src/creator.rs

```diff
@@ -43,7 +43,7 @@ impl<H: Hasher> Creator<H> {
         let n_members = conf.n_members;
         Creator {
             node_ix: conf.node_ix,
-            parents_rx,
+            parents_rx: parents_rx.fuse(),
```

Comment:
I don't think there is a need for that. `Receiver` does implement `FusedStream`: https://docs.rs/futures/0.3.16/futures/channel/mpsc/struct.UnboundedReceiver.html#impl-FusedStream . I dont think the usage of `futures::select!` macro would work if `Receiver` weren't `FusedStream`

Reply:
removed